### PR TITLE
organize requirements files

### PR DESCRIPTION
**Pull request by sfc-gh-pmardziel**
# Description

Organized requirements files for formatting and docs building.

## Other details good to know for developers

- Moved/renamed `docs/docs_requirements.txt` to `requirements.docs.txt`.
- Created yapf/isort installation as new requirements file `requirements.dev.txt`.
- Created test-related package requirements as new file `requirements.test.txt`.
- Updated the uses of these.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

### CODE REVIEW

**Review by sfc-gh-dkurokawa — COMMENTED** [review]
I changed some of this, can you merge from main and Slack me?

File: Makefile [hunk 1]
```diff
@@ -10,7 +10,9 @@ CONDA := source $$(conda info --base)/etc/profile.d/conda.sh ; conda activate ;
 	$(CONDA) .conda/docs; \
 	pip install -r trulens_eval/trulens_eval/requirements.txt; \
 	pip install -r trulens_eval/trulens_eval/requirements.optional.txt; \
-	pip install -r docs/docs_requirements.txt
+	pip install -r requirements.docs.txt; \
```
Comment (sfc-gh-dkurokawa) [inline Makefile:13]: what's with all this `; \` stuff?

**Review by sfc-gh-dkurokawa — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (sfc-gh-pmardziel) [reply Makefile:13]: Its for writing one long command composed of multiple commands (separated by ";") over multiple lines (by escaping newline with "\").

Comment (sfc-gh-dkurokawa) [reply Makefile:13]: What difference is it though if you didn't have it? Like either way it'll run each of the commands one line at a time right?

Comment (sfc-gh-pmardziel) [reply Makefile:13]: I believe by default it does not preserve the environment between each command it launches so each line would have to include the activate conda line.

**Pull request merged**